File: Day 4/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min_index = 0;
        int result = 0;
        int profit;
        for(int i =1; i <prices.size();i++){
            profit = prices[i] - prices[min_index];
            result = max(result, profit);
            if (prices[i] < prices[min_index]){
                min_index = i;
            }
        }
        return result;
    }
};

File: Day 4/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_index = 0
        result = 0
        for i in range(1,len(prices)):
            profit = prices[i] - prices[min_index]
            result = max(result, profit)
            if prices[i] < prices[min_index]:
                min_index = i

        return result
